server/admin_management/tests/services/test_user_validator.py

```diff
@@ -0,0 +1,177 @@
+import logging
+import pytest
+from fastapi import HTTPException
+from mock import patch
+
+from api.app.services.validator.user_validator import UserValidator, IdimProxyService
+from api.app.schemas import Requester, TargetUser
+from api.app.constants import ERROR_CODE_INVALID_REQUEST_PARAMETER, UserType
+from tests.constants import (
+    TEST_USER_NAME_IDIR,
+    TEST_USER_GUID_IDIR,
+    TEST_USER_NAME_BCEID,
+    TEST_USER_GUID_BCEID,
+    TEST_USER_BUSINESS_GUID_BCEID,
+)
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+TEST_NEW_TARGET_USER_IDIR = {
+    "user_type_code": UserType.IDIR,
+    "user_name": TEST_USER_NAME_IDIR,
+    "user_guid": TEST_USER_GUID_IDIR,
+}
+
+TEST_NEW_TARGET_USER_BCEID_LOAD_2 = {
+    "user_type_code": UserType.BCEID,
+    "user_name": TEST_USER_NAME_BCEID,
+    "user_guid": TEST_USER_GUID_BCEID,
+}
+
+TEST_REQUESTER = {
+    "user_id": 11,  # this is a faked id
+    "user_type_code": UserType.IDIR,
+    "user_name": TEST_USER_NAME_IDIR,
+    "user_guid": TEST_USER_GUID_IDIR,
+}
+
+MOCK_SERACH_IDIR_RETURN = {
+    "found": True,
+    "guid": TEST_NEW_TARGET_USER_IDIR.get("user_guid"),
+    "userId": TEST_NEW_TARGET_USER_IDIR.get("user_name"),
+}
+
+MOCK_SERACH_BCEID_RETURN = {
+    "found": True,
+    "guid": TEST_NEW_TARGET_USER_BCEID_LOAD_2.get("user_guid"),
+    "userId": TEST_NEW_TARGET_USER_BCEID_LOAD_2.get("user_name"),
+    "businessGuid": TEST_USER_BUSINESS_GUID_BCEID,
+}
+
+
+class TestUserValidatorClass(object):
+    """
+    Testing UserValidator class with real remote API calls (TEST environment).
```

Comment:
Hi Catherine, we can probably now remove this comment.

Reply:
yeah updated, thanks!